bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3747,57 +3745,70 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
       LOG_DEBUG(CNSUS, "Consensus reached");
     }
     SCOPED_MDC("pp_msg_cid", ppMsg->getCid());
+    std::deque<IRequestsHandler::ExecutionRequest> accumulatedRequests;
     while (reqIter.getAndGoToNext(requestBody)) {
       size_t tmp = reqIdx;
       reqIdx++;
       if (!requestSet.get(tmp)) {
         continue;
       }
-
       ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
       if (req.flags() & EMPTY_CLIENT_FLAG) {
         continue;
       }
       SCOPED_MDC_CID(req.getCid());
       NodeIdType clientId = req.clientProxyId();
-
       uint32_t actualReplyLength = 0;
       uint32_t actualReplicaSpecificInfoLength = 0;
-      int status;
+      int executionStatus = 1;
+      accumulatedRequests.push_back(IRequestsHandler::ExecutionRequest{
+          clientId,
+          static_cast<uint64_t>(lastExecutedSeqNum + 1),
+          req.flags(),
+          std::string(req.requestBuf(), req.requestLength()),
+          std::string(ReplicaConfig::instance().getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader), 0),
```

Comment:
Why do you pre-allocate an empty string? Pass just std::string()

Reply:
I'm allocating with the max response size and then I can use this string size instead of sending also variable that holds the max response size.

Reply:
We saw that no matter what at some point the max size will be allocated so it's ok to leave it like that.